docs/articles/documentation/guides/basic-guides/obtain-client-side-info.md

```diff
@@ -0,0 +1,173 @@
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const performAsyncOperation = ClientFunction(() => {
+    return new Promise(resolve => {
+        window.setTimeout(resolve, 500); // some async operations
+    });
+});
+```
+
+## Execute Client Functions
+
+Call the client function with the `await` keyword to execute it.
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const getWindowLocation = ClientFunction(() => window.location);
+
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    const location = await getWindowLocation();
+});
+```
+
+## Overwrite Client Function Option
+
+You can overwrite client function options with the ClientFunction's [with method](../../reference/test-api/clientfunction/with.md).
+
+```js
+const cfWithDependency = cfWithoutDependency.with({
+    dependencies: { foo: 'bar' }
+});
+```
+
+## One-Time Client Code Execution
+
+You can use the [eval method](../../reference/test-api/testcontroller/eval.md) of the [test controller](../../reference/test-api/testcontroller/README.md) to create a client function and execute it immediately without saving it.
+
+```js
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    const docURI = await t.eval(() => document.documentURI);
+});
+```
+
+## Import Functions to be Used as Client Function Dependencies
+
+Assume you have a JS file `utils.js` with a function you need to use as a client function dependency in your test file.
+
+**utils.js**
+
+```js
+export function getDocumentURI() {
+    return document.documentURI;
+}
+```
+
+Note that TestCafe processes test files with [Babel](https://babeljs.io) internally. To avoid issues caused by code transpiling, use the [require](https://nodejs.org/api/modules.html#modules_require_id) function instead of the `import` statement to import client function dependencies.
+
+**test.js**
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const getDocumentURI = require('./utils.js').getDocumentURI;
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const getUri = ClientFunction(() => {
+        return getDocumentURI();
+    }, { dependencies: { getDocumentURI } });
+
+    const uri = await getUri();
+
+    await t.expect(uri).eql('http://devexpress.github.io/testcafe/example/');
+});
+```
+
+## Call Client Functions from Node.js Callbacks
+
+Client functions need access to the [test controller](../../reference/test-api/testcontroller/README.md) to be executed.
+When called right from the test function, they implicitly obtain the test controller.
+
+However, if you need to call a client function from a Node.js callback that fires during the test run,
+you need to manually bind this function to the test controller.
```

Comment:
Change to:
Is there a better word to use than "during" ("during the test run")? - to avoid the gerund. For example, if the callback fires at the start of the test run, could we say "upon test execution". Please clarify.